lib/yao/resources/user.rb

```diff
@@ -9,8 +9,11 @@ class User < Base
     self.resources_name = "users"
     self.admin          = true
 
-    def self.get_by_name(name)
-      self.list.find {|user| user.name == name }
+    class << self
+      def get_by_name(name)
+        self.get("", name: name)
```

Comment:
This will make URL with a trailing slash,  like `/users/?name=foo`

How about using `self.list(name: name).first` ?